-Wdeprecated-copy-dtor warnings

Using release version 10.4.0 and Clang 11 (C++17 build) on Linux, I get the following `-Wdeprecated-copy-dtor` warnings:

```
In file included from ../external/approvaltests.cpp/ApprovalTests/namers/../Approvals.h:13:
../external/approvaltests.cpp/ApprovalTests/namers/DefaultNamerDisposer.h:17:9: error: definition of implicit copy constructor for 'DefaultNamerDisposer' is deprecated because it has a user-declared destructor [-Werror,-Wdeprecated-copy-dtor]
../external/approvaltests.cpp/ApprovalTests/namers/SeparateApprovedAndReceivedDirectoriesNamer.cpp:32:16: note: in implicit copy constructor for 'ApprovalTests::DefaultNamerDisposer' first required here
```

```
In file included from ../external/approvaltests.cpp/ApprovalTests/reporters/CIBuildOnlyReporterUtils.cpp:1:
In file included from ../external/approvaltests.cpp/ApprovalTests/../ApprovalTests/reporters/CIBuildOnlyReporterUtils.h:4:
../external/approvaltests.cpp/ApprovalTests/../ApprovalTests/reporters/FrontLoadedReporterDisposer.h:17:9: error: definition of implicit copy constructor for 'FrontLoadedReporterDisposer' is deprecated because it has a user-declared destructor [-Werror,-Wdeprecated-copy-dtor]
../external/approvaltests.cpp/ApprovalTests/reporters/CIBuildOnlyReporterUtils.cpp:10:16: note: in implicit copy constructor for 'ApprovalTests::FrontLoadedReporterDisposer' first required here
```

Let me know if you need further details.

--- a/ApprovalTests/namers/DefaultNamerDisposer.h
+++ b/ApprovalTests/namers/DefaultNamerDisposer.h
@@ -13,6 +13,7 @@ namespace ApprovalTests
 
     public:
         explicit DefaultNamerDisposer(NamerCreator namerCreator);
+        DefaultNamerDisposer(const DefaultNamerDisposer&) = default;
 
         ~DefaultNamerDisposer();
     };

--- a/ApprovalTests/namers/SectionNameDisposer.h
+++ b/ApprovalTests/namers/SectionNameDisposer.h
@@ -9,6 +9,7 @@ namespace ApprovalTests
     {
     public:
         SectionNameDisposer(TestName& currentTest_, const std::string& scope_name);
+        SectionNameDisposer(const SectionNameDisposer&) = default;
 
         ~SectionNameDisposer();
 

--- a/ApprovalTests/namers/SubdirectoryDisposer.h
+++ b/ApprovalTests/namers/SubdirectoryDisposer.h
@@ -14,6 +14,7 @@ namespace ApprovalTests
 
     public:
         explicit SubdirectoryDisposer(std::string subdirectory);
+        SubdirectoryDisposer(const SubdirectoryDisposer&) = default;
 
         ~SubdirectoryDisposer();
     };

--- a/ApprovalTests/reporters/FrontLoadedReporterDisposer.h
+++ b/ApprovalTests/reporters/FrontLoadedReporterDisposer.h
@@ -13,6 +13,7 @@ namespace ApprovalTests
 
     public:
         explicit FrontLoadedReporterDisposer(const std::shared_ptr<Reporter>& reporter);
+        FrontLoadedReporterDisposer(const FrontLoadedReporterDisposer&) = default;
 
         ~FrontLoadedReporterDisposer();
     };

